exonum/src/blockchain/mod.rs

```diff
@@ -183,12 +184,53 @@ impl Blockchain {
         let has_genesis_block = !Schema::new(&self.snapshot())
             .block_hashes_by_height()
             .is_empty();
-        if !has_genesis_block {
+        if has_genesis_block {
+            self.check_storage_version();
+        } else {
+            self.initialize_metadata();
```

Comment:
Would it make sense to move `initialize_metadata` invocation into `create_genesis_block`? IMO, initialization logically belongs there.

Reply:
This got me wondering as well. My thoughts were that we write data to the storage with some specific version. Otherwise it would be that genesis block is written before the storage has a version.